### DISCUSSION TITLE
Help with heic files - thumbnails and viewing

### DISCUSSION THREAD

**Discussion by qpkorr** [doublecmd/doublecmd#1749] [General]
Hi - thanks very much, doublecmd is awesome!

One small issue I'm having - I'm not getting thumbnails for heic files, and when I try to view them, the text viewer opens.

The versions I'm using, on up to date Ubuntu 23.10:
ii  doublecmd-qt                                         1.1.16-0+svn520~testing
ii  doublecmd-common                               1.1.16-0+svn520\~testing
ii  doublecmd-plugins                              1.1.16-0+svn520\~testing)
ii  heif-gdk-pixbuf:amd64                          1.16.2-2ubuntu1.1
ii  heif-thumbnailer                               1.16.2-2ubuntu1.1
ii  libheif-examples                               1.16.2-2ubuntu1.1
ii  libheif-plugin-aomenc:amd64                    1.16.2-2ubuntu1.1
ii  libheif-plugin-dav1d:amd64                     1.16.2-2ubuntu1.1
ii  libheif-plugin-libde265:amd64                  1.16.2-2ubuntu1.1
ii  libheif-plugin-x265:amd64                      1.16.2-2ubuntu1.1
ii  libheif1:amd64                                 1.16.2-2ubuntu1.1

My understanding is that at least some of these libraries would have been enough?  Thanks for any tips!
PS - thumbnails are showing up correctly on Nautilus

**Comment by alexx2000** [comment]
Hi, `libheif1` is enough. Try to execute DC from terminal, maybe some error will be shown.

**Reply by qpkorr** [reply]
Thanks! Yep, there's an error:
`#: john@j4lt:~ ; doublecmd
SystemLocale en_AU
SystemLanguage en
SystemEncoding utf8
DefaultSystemCodePage 65001
DefaultFileSystemCodePage 65001
DefaultRTLFileSystemCodePage 65001
ImageMagick: libMagickWand-6.Q16.so.6
Starting GuiMessageQueue
Starting Double Commander
Double Commander 1.1.16 gamma
Revision: 520
Commit: ca77fea98
Build: 2024/06/22
Lazarus: 2.2.6.0
Free Pascal: 3.2.2
Platform: x86_64-Linux-qt5
System: Ubuntu 23.10
Desktop Environment: Unknown
Widgetset library: Qt 5.15.10, libQt5Pas 5.6.1
This program is free software released under terms of GNU GPL 2
Copyright (C) 2006-2024 Alexander Koblov
   and contributors (see about dialog)
Executable directory: /usr/lib/doublecmd/
Configuration directory: /home/john/.config/doublecmd
Global configuration directory: /usr/lib/doublecmd/settings/
Loading configuration from /home/john/.config/doublecmd/
Error loading configuration file /home/john/.config/doublecmd/favoritetabs.xml
PollThread: Start polling
PollThread: AddPoll 17
Creating PixmapManager
Loading icon theme Hicolor
Loading icon theme DCTheme
TheFontStock.CalcFontAdvance: Font=courier Size=12 Width(8) > tmMaxWidth+Over(5)
TColumnsFileView.Create components
Creating TFileSystemFileSource
FileSystemWatcher thread starting
PollThread: AddPoll 24
UDev: Begin monitoring
Detecting mounts through /proc/self/mounts
PollThread: AddPoll 25
frmMain.frmMainShow`
`Error: Cannot load pixmap [/mnt/photo/john/backup/20240714_140226.heic] : Unsupported feature: Unsupported codec`
`Error: Cannot load pixmap [/mnt/photo/john/backup/20240714_140220.heic] : Unsupported feature: Unsupported codec`
`Error: Cannot load pixmap [/mnt/photo/john/backup/20240714_140215.heic] : Unsupported feature: Unsupported codec`
`Error: Cannot load pixmap [/mnt/photo/john/backup/20240714_140156.heic] : Unsupported feature: Unsupported codec`
`Error: Cannot load pixmap [/mnt/photo/john/backup/20240714_140145.heic] : Unsupported feature: Unsupported codec`
`Error: Cannot load pixmap [/mnt/photo/john/backup/20240714_140137.heic] : Unsupported feature: Unsupported codec`

Given the "pix" in there I tried removing heif-gdk-pixbuf - same error.  I seem to recall adding some of the other libraries to add support for HEIC/F for other software - maybe gimp, imagemagick, image viewers, video players (for AV1) - I'm not sure - but I guess I can try removing all of them except libheif1 and see how I go - at least as a start.

I removed heif-gdk-pixbuf, heif-thumbnailer, libheif-examples, libheif-plugin-aomenc to no avail, same error.  When I tried to remove libheif-plugin-dav1d apt wanted to reinstall libheif-plugin-aomdec - so I gave up at this point.  Tried reinstalling libheif1 (apt install --reinstall) - no luck.

Any more suggestions?  Perhaps I could check source code for that error message, but which source code...?  Thanks!

**Comment by alexx2000** [comment]
Can you give a some sample image which does not work?

**Reply by qpkorr** [reply]
Anything from my Samsung S24 Ultra seems to do it, eg
[20240714_140113.gz](https://github.com/user-attachments/files/16225331/20240714_140113.gz)
Rename that to .heic - github doesn't support heic either!

**Comment by alexx2000** [comment]
Looks like `libheif1` built without some codec support in Ubuntu 23.10. I opened your sample image under my Arch Linux and it [works](https://imgur.com/PfWZtPO). Also I tried Ubuntu 23.10 in Virtual Box and it does not work.

**Comment by alexx2000** [comment]
I tried Ubuntu 24.04 in Virtual Box and it works. It has a newer `libheif` version.

**Reply by qpkorr** [reply]
Thanks so much!!! I was planning to wait for 24.04.1 as I gather that's recommended for LTS people like me - sounds like I should get 24.04 now afterall. Amazing support, thanks so much!
